File: THCS/2022_2023/HauGiang_2223/Bai1/Cau1.py
def main():
    # Đọc dữ liệu từ tệp xau.inp
    with open("xau.inp", "r") as input_file:
        s = input_file.readline().strip()

    # Tạo một tập hợp chứa tất cả các ký tự từ A đến Z
    all_characters = set("ABCDEFGHIJKLMNOPQRSTUVWXYZ")

    # Tạo một tập hợp từ xâu S
    present_characters = set(s)

    # Tìm các ký tự còn thiếu
    missing_characters = sorted(all_characters - present_characters)

    # Ghi ra tệp xau.out các ký tự còn thiếu
    with open("xau.out", "w") as output_file:
        output_file.write("".join(missing_characters))

if __name__ == "__main__":
    main()

File: THCS/2022_2023/HauGiang_2223/Bai1/Cau1.cpp
#include <iostream>
#include <fstream>
#include <set>
#include <algorithm>

int main() {
    // Đọc dữ liệu từ tệp xau.inp
    std::ifstream inputFile("xau.inp");
    std::string s;
    inputFile >> s;
    inputFile.close();

    // Tạo một tập hợp chứa tất cả các ký tự từ A đến Z
    std::set<char> allCharacters;
    for (char c = 'A'; c <= 'Z'; ++c) {
        allCharacters.insert(c);
    }

    // Tạo một tập hợp chứa các ký tự đã xuất hiện trong xâu S
    std::set<char> presentCharacters(s.begin(), s.end());

    // Tìm các ký tự còn thiếu
    std::set<char> missingCharacters;
    for (char c : allCharacters) {
        if (presentCharacters.find(c) == presentCharacters.end()) {
            missingCharacters.insert(c);
        }
    }

    // Ghi ra tệp xau.out các ký tự còn thiếu
    std::ofstream outputFile("xau.out");
    for (char c : missingCharacters) {
        outputFile << c;
    }
    outputFile.close();

    return 0;
}
